### PR TITLE
feat: event processing in proxy

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of #378 

Added proxy repository processing, workflow is the following:
1. Proxy repository adds npm package (tgz archive) storage key into queue after the package was downloaded from remote
2. Job check the packages from queue: 
  a) that tgz is valid
  b) corresponding json metadata file exists, contains tgz version and path to this tgz.
3. Then adds package info to package queue to be added database

This job can be also used to index some new repository and fill packages database. 

Build is failing as not existing http dependency is used (will correct it as soon as we release http).

### CODE REVIEW

File: src/main/java/com/artipie/npm/http/UnpublishForceSlice.java [hunk 1]
```diff
@@ -83,8 +84,10 @@ public Response response(
             ).value();
             resp = new AsyncResponse(
                 this.storage.deleteAll(new Key.From(pkg)).thenAccept(
-                    nothing -> this.events.put(
-                        new ArtifactEvent(UploadSlice.REPO_TYPE, this.rname, pkg)
+                    nothing -> this.events.map(
```
Comment (dgarus) [inline src/main/java/com/artipie/npm/http/UnpublishForceSlice.java:87]: We don't need to create a future if `events` is empty.
This comment is valid for all places like this.

**Comment by olenagerasimova** [comment]
@dgarus please, review

File: src/main/java/com/artipie/npm/events/NpmProxyPackageProcessor.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+ * <br/>
+ * 1) tgz archive is valid and we obtained package id and version from it<br/>
+ * 2) repository has corresponding package json metadata file with such version and
+ *   path to tgz
+ * <br/>
+ * When both conditions a met, we can add package record into database.
+ * @since 1.5
+ */
+@SuppressWarnings("PMD.DataClass")
+public final class NpmProxyPackageProcessor extends QuartzJob {
```
Comment (dgarus) [inline src/main/java/com/artipie/npm/events/NpmProxyPackageProcessor.java:35]: I didn't find the place when ` NpmProxyPackageProcessor` is used.
Could you please explain how we plan to write events to the DB?

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks. Please look at my comments.

File: src/main/java/com/artipie/npm/events/NpmProxyPackageProcessor.java (hunk 2 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/npm/events/NpmProxyPackageProcessor.java:35]: The job will be started from artipie if artifact db will be configured in repository settings

Comment (dgarus) [reply src/main/java/com/artipie/npm/events/NpmProxyPackageProcessor.java:35]: Ok

**Review by dgarus — APPROVED** [review]
lgtm

**Pull request merged**